### PR TITLE
Documentation for `WiFi.setHostname()` and `WiFi.hostname()`

### PR DESCRIPTION
**Pull request by avtolstoy**
Adds documentation for features introduced in spark/firmware#1296

### CODE REVIEW

File: src/content/reference/firmware.md [hunk 1]
```diff
@@ -1433,6 +1433,49 @@ A note on switching between static and dynamic IP. If static IP addresses have b
...
+Serial.println(System.deviceID());
+```
+
+### hostname()
+
+_Since 0.7.0_
+
+Retrieves device hostname used as DHCP client name (DHCP option 12).
+
+This function does not take any arguments and returns a `String`.
```
Comment (kennethlimcp) [inline src/content/reference/firmware.md:1470]: Argument

**Review by kennethlimcp — COMMENTED** [review]
Is this implemented for `core`?

**Comment by avtolstoy** [comment]
@kennethlimcp Photon/P1 only. The documentation for these functions is within `{{#if photon}}` block.

**Comment by kennethlimcp** [comment]
Perfect :)

**Review by technobly — APPROVED** [review]

**Pull request merged**